MapAdjuster/Source/Frames/PointWindow.cpp

```diff
@@ -27,3 +27,100 @@ PointWindow::PointWindow(wxWindow* parent): wxWindow(parent, wxID_ANY, wxDefault
 	SetSizer(logBox);
 	logBox->Fit(this);
 }
+
+void PointWindow::redrawGrid()
+{
+	auto rowCounter = 0;
+	theGrid->BeginBatch();
+	theGrid->DeleteRows(0, static_cast<int>(coPoints->size()));
+	for (const auto& coPoint: *coPoints)
+	{
+		std::string name;
+		if (coPoint->getName().empty())
+			name = "(No Name)";
+		else
+			name = coPoint->getName();
+		std::string source;
+		if (coPoint->getSource())
+			source = std::to_string(coPoint->getSource()->x) + "x" + std::to_string(coPoint->getSource()->y);
+		std::string target;
+		if (coPoint->getTarget())
+			target = std::to_string(coPoint->getTarget()->x) + "x" + std::to_string(coPoint->getTarget()->y);
+
+		const auto bgcolor = wxColour(200, 200, 200);
+
+		theGrid->AppendRows(1, false);
+		theGrid->SetRowSize(rowCounter, 20);
+		theGrid->SetCellValue(rowCounter, 0, name);
+		theGrid->SetCellAlignment(rowCounter, 0, wxCENTER, wxCENTER);
+		theGrid->SetReadOnly(rowCounter, 0);
+		theGrid->SetCellValue(rowCounter, 1, source);
+		theGrid->SetCellBackgroundColour(rowCounter, 1, bgcolor);
+		theGrid->SetCellAlignment(rowCounter, 1, wxCENTER, wxCENTER);
+		theGrid->SetReadOnly(rowCounter, 1);
+		theGrid->SetCellValue(rowCounter, 2, target);
+		theGrid->SetCellBackgroundColour(rowCounter, 2, bgcolor);
+		theGrid->SetCellAlignment(rowCounter, 1, wxLEFT, wxCENTER);
+		theGrid->SetReadOnly(rowCounter, 2);
+		rowCounter++;
+	}
+	theGrid->EndBatch();
+	theGrid->AutoSize();
+	GetParent()->Layout();
+	theGrid->Scroll(0, rowCounter - 1);
+}
+
+void PointWindow::registerPoint(const Point& point, ImageTabSelector theSelector)
+{
+	wxCommandEvent evt(wxEVT_UPDATE_POINT);
+	if (workingPoint)
```

Comment:
This block is long enough on screen and nests deep enough that it might be worth creating functions to call for clarification